pkg/controller/rollout/rollout_controller.go

```diff
@@ -79,6 +90,13 @@ func (r *RolloutReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, err
 	}
 
+	watched, err := util.AddDynamicWatcher(runtimeController, workloadHandler, watchedSet, rollout.Spec.ObjectRef.WorkloadRef)
+	if err != nil {
+		return ctrl.Result{}, err
+	} else if watched {
+		watchedSet.Insert(util.GetGVKFrom(rollout.Spec.ObjectRef.WorkloadRef).String())
```

Comment:
shall we wait for informer cache syncd before continue reconcile?

Reply:
done